File: Cap1VB/ProcessosAntesLINQ.vb
Imports System.Collections.Generic
Imports System.Diagnostics

Public Class ProcessosAntesLINQ
  Public Function ImprimeProcessos() As IEnumerable(Of Process)
    Dim processos As New List(Of Process)()
    For Each p As Process In Process.GetProcesses()
      If p.ProcessName.StartsWith("w") Then
        processos.Add(p)
      End If
    Next
    Return processos
  End Function
End Class

File: Cap1CSharp/ProcessosAntesLINQ.cs
using System.Collections.Generic;
using System.Diagnostics;

namespace Cap1CSharp 
{
    public class ProcessosAntesLINQ 
    {
        public IEnumerable<Process> ImprimeProcessos()
        {
            List<Process> processos = new List<Process>();
            foreach (Process p in Process.GetProcesses()) 
            {
                if (p.ProcessName.StartsWith("w")) 
                {
                    processos.Add(p);
                }
            }
            return processos;
        }
    }
}
